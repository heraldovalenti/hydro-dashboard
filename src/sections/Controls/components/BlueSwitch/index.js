import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import COLORS from '../../../../types/colors';
import { Switch } from '@material-ui/core';

const BlueSwitch = withStyles({
  switchBase: {
    color: '#fafafa',
    '&$checked': {
      color: COLORS.TURQUOISE.BASE,
    },
    '&$checked + $track': {
      backgroundColor: COLORS.TURQUOISE.BASE,
    },
  },
  checked: {},
  track: {},
})(Switch);

export default BlueSwitch;
