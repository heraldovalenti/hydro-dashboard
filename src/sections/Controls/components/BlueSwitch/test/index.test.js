import React from 'react';
import BlueSwitch from '../';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('<BlueSwitch/>', () => {
  it('default', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<BlueSwitch />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
