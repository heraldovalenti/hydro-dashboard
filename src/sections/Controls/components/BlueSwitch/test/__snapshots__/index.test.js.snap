// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlueSwitch/> default 1`] = `
<ForwardRef(WithStyles)
  classes={
    Object {
      "checked": "WithStyles(ForwardRef(Switch))-checked-2",
      "switchBase": "WithStyles(ForwardRef(Switch))-switchBase-1",
      "track": "WithStyles(ForwardRef(Switch))-track-3",
    }
  }
/>
`;
